presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/dateformat/DateFormatParser.java

```diff
@@ -40,28 +70,29 @@ public static DateTimeFormatter createDateTimeFormatter(String format)
                     builder.appendLiteral(token.getText());
                     break;
                 case DateFormat.DD:
-                    builder.appendDayOfMonth(2);
+                    builder.appendValue(ChronoField.DAY_OF_MONTH, mode.minTwoPositionFieldWidth, 2, SignStyle.NOT_NEGATIVE);
                     break;
                 case DateFormat.HH24:
-                    builder.appendHourOfDay(2);
+                    builder.appendValue(ChronoField.HOUR_OF_DAY, mode.minTwoPositionFieldWidth, 2, SignStyle.NOT_NEGATIVE);
                     break;
                 case DateFormat.HH:
-                    builder.appendHourOfHalfday(2);
+                    // ChronoField.HOUR_OF_AMPM should be used for HH after format extended to include AM/PM token
```

Comment:
Is this comment still valid ?

Reply:
At the moment format does not allow to include AM/PM token, thus it was never possible to specify f.e. PM hours using 'HH' token in format. In 2b89c16 added more explanation on the matter.